pkg/frontend/openshiftcluster_list.go

```diff
@@ -64,5 +66,49 @@ func (f *frontend) _getOpenShiftClusters(ctx context.Context, r *http.Request, c
 		ocs[i].Properties.ServicePrincipalProfile.ClientSecret = ""
 	}
 
-	return json.MarshalIndent(converter.ToExternalList(ocs), "", "    ")
+	nextLink, err := addSkipToken(r.Header.Get("Referer"), i.Continuation())
+	if err != nil {
+		return nil, err
+	}
+
+	return json.MarshalIndent(converter.ToExternalList(ocs, nextLink), "", "    ")
+}
+
+// getSkipToken parses originalURL for skipToken.
+// Returns and empty string without an error, if there is no $skipToken in the URL
+func getSkipToken(originalURL string) (string, error) {
+	u, err := url.Parse(originalURL)
+	if err != nil {
+		return "", err
+	}
+
+	query, err := url.ParseQuery(u.RawQuery)
```

Comment:
`r.URL.Query().Get("$skipToken")` doesn't do all this for you?

Reply:
Argh! I did look for `Query` in the `url.URL`, but didn't notice it.